### ISSUE TITLE
Vertx.close().setHandler() the handler is never do, because of the eventloop is shutdown.

### ISSUE THREAD

**Issue by takeseem** [vert-x3/issues#521]
### Questions

```
CompletableFuture<T> cf = new CompletableFuture<>();
vertx.close().setHandler(v -> {
	if (v.succeeded()) {
		cf.complete(v.result());
	} else {
		cf.completeExceptionally(v.cause());
	}
});
try {
	cf.get();
} catch (Throwable t) {
	logger.warn("close: {}", vertx, t);
}
// do other something, but it's never todo here
```
the `Future<Void> Vertx.close()` must be Deprecated, because the return Future never handle, the eventloop is shutdown when Vertx.close().

my right code is use `close(Handler)`, it's worked. 
```
CompletableFuture<T> cf = new CompletableFuture<>();
vertx.close(v -> {
	if (v.succeeded()) {
		cf.complete(v.result());
	} else {
		cf.completeExceptionally(v.cause());
	}
});
try {
	cf.get();
} catch (Throwable t) {
	logger.warn("close: {}", vertx, t);
}
```

### Version

4.0.0-milestone4

### Context

```
Exception in thread "vert.x-eventloop-thread-16" java.util.concurrent.RejectedExecutionException: event executor terminated
	at io.netty.util.concurrent.SingleThreadEventExecutor.reject(SingleThreadEventExecutor.java:981)
	at io.netty.util.concurrent.SingleThreadEventExecutor.offerTask(SingleThreadEventExecutor.java:388)
	at io.netty.util.concurrent.SingleThreadEventExecutor.addTask(SingleThreadEventExecutor.java:381)
	at io.netty.util.concurrent.SingleThreadEventExecutor.execute(SingleThreadEventExecutor.java:880)
	at io.vertx.core.impl.EventLoopContext.execute(EventLoopContext.java:42)
	at io.vertx.core.impl.EventLoopContext.dispatch(EventLoopContext.java:77)
	at io.vertx.core.impl.EventLoopContext.dispatch(EventLoopContext.java:65)
	at io.vertx.core.impl.FutureImpl.doDispatch(FutureImpl.java:125)
	at io.vertx.core.impl.FutureImpl.dispatch(FutureImpl.java:119)
	at io.vertx.core.impl.FutureImpl.tryComplete(FutureImpl.java:149)
	at io.vertx.core.Promise.complete(Promise.java:67)
	at io.vertx.core.Promise.handle(Promise.java:52)
	at io.vertx.core.Promise.handle(Promise.java:30)
	at io.vertx.core.impl.VertxImpl$1$1.lambda$operationComplete$0(VertxImpl.java:919)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:830)
```

### Extra

* Ubuntu 18.04.4 LTS
* OpenJDK 64-Bit Server VM (build 14+36-1461, mixed mode, sharing)

**Comment by BillyYccc** [comment]
I think this is duplicate of https://github.com/eclipse-vertx/vert.x/issues/3232 and has been fixed in `master` branch, can you try with `4.0.0-SNAPSHOT`?

**Comment by takeseem** [comment]
It's fix in `4.0.0-SNAPSHOT` . 
Thanks.
